core/storage/btree.rs

```diff
@@ -5108,6 +5193,24 @@ impl BTreeCursor {
                     //  Get the current cell
                     let cell = contents.cell_get(cell_idx as usize, self.usable_space())?;
 
+                    // Count cells on all leaf pages (both table and index leaves)
```

Comment:
Index interior pages also have payloads - does SQLite not count them here?

Reply:
No, SQLite does not count them - or I misunderstood the check. The best thing if you suspect that is to come up with a test case where the BTree is relatively large, and see what sqlite and turso return. I'll check that.

Reply:
yes when its a table interior key, not an index interior key

Reply:
Again, what am I missing here? Isn't the test operating on an index btree?

Reply:
should be good now.